### PR TITLE
feat(storybook): cleanup volume 1

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

We're at a place where lack of standardised components within App is harming velocity. We have Storybook for this use, but it's underutilised due to various issues. This is the first PR in a series of PRs to clean up storybook, and fix the broken mocking strategy.

As always, I'd love to get this in fast to have something to build upon :).

## Changes

Many changes.

- Reorder the stories into this list. It's not perfect, but it's a step in the right direction. Let's do changes in follow up PRs.
image

- Move `.storybook/preview.ts` to TypeScript, alias the storybook folder as `storybook/` so we could import from there, move the `kea-story.tsx` file here. 
- Wrap each story with `<KeaStory>`, no questions asked. All logic should just work everywhere.
- Share the same mock service worker system with jest tests, this includes sharing default mocks with jest. This means things generally just load, if with empty data. 
- New boilerplate-less way of mocking API requests for specific stories, plus converted two stories to use this:
   - frontend/src/lib/components/EventSelect/EventSelect.stories.tsx
   - frontend/src/lib/components/HistoryList/__stories__/HistoryList.stories.tsx

```ts
// Something.stories.tsx
export default {
    title: 'Filters/EventSelect',
    decorators: [
        mswDecorator({ // either use a global decorator
            get: {
                '/api/projects/:projectId/event_definitions': {
                    count: eventDefinitions.length,
                    next: null,
                    previous: null,
                    results: eventDefinitions,
                },
            },
        }),
    ],
} as Meta

export const WithData = (): JSX.Element => {
    // or mock per story
    useStorybookMocks({
        get: {
            '/api/projects/@current/feature_flags/7/history': { results: featureFlagsHistoryResponseJson },
        },
    })

    return <HistoryList type={'FeatureFlag'} id={7} />
}
```

## How did you test this code?

Storybook still runs. There are still many broken stories, and I'd love to fix them all in the next PRs. Nothing new should be broken.

### CODE REVIEW

File: frontend/src/lib/components/EventSelect/EventSelect.stories.tsx [hunk 1]
```diff
@@ -57,39 +36,20 @@ export const Default = (): JSX.Element => {
...
+        }),
+    ],
+} as Meta
 
-    // Need to mount teamLogic otherwise we get erros regarding not being able
-    // to find the storre for team. It also makes some API calls to
-    // `/api/projects/@current` and `/api/organizations/@current` although I
-    // haven't mocked these as the component still works without doing so
-    teamLogic.mount()
+export const Default = (): JSX.Element => {
```
Comment (pauldambra) [inline frontend/src/lib/components/EventSelect/EventSelect.stories.tsx:45]: fantastic amount of boilerplate disappearing!

**Review by pauldambra — APPROVED** [review]
yep, a few broken stories. But lots of boiler plate removed and makes it easier to add/edit stories 🆗

**Review by hazzadous — APPROVED** [review]
Looks good

Just a non-blocker re. defining handlers.

File: frontend/src/lib/components/EventSelect/EventSelect.stories.tsx [hunk 2]
```diff
@@ -57,39 +36,20 @@ export const Default = (): JSX.Element => {
...
+            addElement={<span>add events</span>}
+        />
     )
 }
-
-type GetEventDefinitionsResponse = EventDefinitionStorage
-type GetEventDefinitionsRequest = undefined
-
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
-const mockGetEventDefinitions = (
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/lib/components/EventSelect/EventSelect.stories.tsx:89]: Absolutely!  I want to eventually have these three modes that you can use:

```ts
mswDecorator({
    get: {
        '/api/projects/:projectId': { id: 2 },
        '/api/projects/:projectId/custom-status': () => [200, { id: 2 }],
        '/api/projects/:projectId/custom-everything': (req, res, ctx) => [ctx.delay(100), ctx.status(200), ctx.json({ id: 2 })],
    },
}),
```

Currently just the first two work, and we're just one `if` away from making the third option functional.

**Pull request merged**